File: asp-core/MB.CityCenter/MB.CityCenter.Dtos/Orders/OrderDetailsDto.cs
using MB.CityCenter.Dtos.Products;
using MB.CityCenter.Utils.Enums;

namespace MB.CityCenter.Dtos.Orders
{
    public class OrderDetailsDto
    {
        public int Id { get; set; }
        public DateTime? Date { get; set; }
        public OrderStatus OrderStatus { get; set; } = OrderStatus.Pending;
        public decimal TotalPrice { get; set; }
        public PaymentMethod PaymentMethod { get; set; } = PaymentMethod.Cash;
        public string CustomerFullName { get; set; }

        public List<SimpleProductDto> Products { get; set; } = new List<SimpleProductDto>();
    }
}

File: angular/src/app/dtos/orders/order-details.dto.ts
import { OrderStatus } from "src/app/enums/order-status.enum";
import { PaymentMethod } from "src/app/enums/payment-method.enum";
import { SimpleProductDto } from "../product/simple-product.model";

export interface OrderDetailsDto {
  id: number;
  date: string;
  orderStatus: OrderStatus;
  totalPrice: number;
  paymentMethod: PaymentMethod;
  customerFullName: string;
  products: SimpleProductDto[];
}
